### PR TITLE
Oauth

### PR DESCRIPTION
**Pull request by Ponnarasi-TechConative**

### CODE REVIEW

File: src/core/components/ConfigModel.tsx [hunk 1]
```diff
@@ -266,7 +291,11 @@ export default function ConfigModel({ handleOpen, handleClose, handleParentModal
                                     label={translate('CALLBACK') + ' ' + translate('URL')}
                                     placeholder={translate('CALLBACK') + ' ' + translate('URL')}
                                 />
-                                <Tooltip onMouseLeave={handleTooltipMouseLeave} onClick={() => handleCopyClick(`https://www.wavemakeronline.com/studio/services/oauth2/${providerConf?.providerId}/callback`)} sx={{ ":hover": { backgroundColor: 'transparent' } }} title={tooltipTitle}>
+                                <Tooltip onMouseLeave={handleTooltipMouseLeave} onClick={() => handleCopyClick(providerConf
+                                    ? `https://www.wavemakeronline.com/studio/services/oauth2/${providerConf.providerId}/callback`
```
Comment (vivek-dhayalan) [inline src/core/components/ConfigModel.tsx:295]: can we move the domain name to configuration?

File: src/core/components/ConfigModel.tsx [hunk 2]
```diff
@@ -266,7 +291,11 @@ export default function ConfigModel({ handleOpen, handleClose, handleParentModal
...
+                                    : providerId
+                                        ? `https://www.wavemakeronline.com/studio/services/oauth2/${providerId}/callback`
```
Comment (vivek-dhayalan) [inline src/core/components/ConfigModel.tsx:297]: +!

File: src/core/components/ConfigModel.tsx [hunk 3]
```diff
@@ -266,7 +291,11 @@ export default function ConfigModel({ handleOpen, handleClose, handleParentModal
...
+                                        : `https://www.wavemakeronline.com/studio/services/oauth2/{providerId}/callback`)} sx={{ ":hover": { backgroundColor: 'transparent' } }} title={tooltipTitle}>
```
Comment (vivek-dhayalan) [inline src/core/components/ConfigModel.tsx:298]: +1

File: src/core/components/appStore/Slice.tsx [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+interface CommonState {
+    selectedProvider: ProviderI;
+    providerAuthURL: string;
+    providerList: [];
+    configOpen: boolean,
+    providerOpen: boolean
+}
+
+const commonSlice = createSlice({
```
Comment (vivek-dhayalan) [inline src/core/components/appStore/Slice.tsx:12]: Any reason on why we have it as slice?

File: README.md [hunk 5]
```diff
@@ -1,51 +1,31 @@
...
-Open [http://localhost:3800](http://localhost:3800) to view it in the browser.
+### Steps
 
-The page will reload if you make edits.\
-You will also see any lint errors in the console.
+1. `git clone https://github.com/wavemaker/rest-client-ui.git`
+2. `cd rest-client-ui`
+3. `npm install`
+4. `npm start`
+5. Wait a bit
```
[the remark this answers is not in the document]
Comment (Ponnarasi-TechConative) [reply README.md]: Removed steps

File: examples/angular-app/angular.json [hunk 6]
```diff
@@ -94,7 +95,9 @@
               "src/assets/plugins/rest-import-bundle.css"
             ],
             "scripts": [
-              "src/assets/plugins/rest-import-bundle.js"
+              "src/assets/plugins/rest-import-bundle.js",
+              "src/assets/plugins/config-import.bundle.js"
```
[the remark this answers is not in the document]
Comment (Ponnarasi-TechConative) [reply examples/angular-app/angular.json:99]: Yes Vivek. CSS file common for both rest Import and Config import

File: examples/angular-app/src/app/fullscreen/fullscreen.component.ts [hunk 7]
```diff
@@ -30,6 +31,6 @@ export class FullscreenComponent implements OnInit {
             "/projects/WMPRJ2c91808888f5252401896880222516b1/oauth2/providers",
```
[the remark this answers is not in the document]
Comment (Ponnarasi-TechConative) [reply examples/angular-app/src/app/fullscreen/fullscreen.component.ts]: Removed the hard coded project id

File: src/core/components/ConfigModel.tsx [hunk 8]
```diff
@@ -10,14 +10,18 @@ import { Alert, Checkbox, DialogActions, FormControl, FormControlLabel, Grid, Ic
...
+import { setProviderAuthorizationUrl, setSelectedProvider, setproviderList } from './appStore/Slice';
+import { useDispatch, useSelector } from 'react-redux';
+import FallbackSpinner from './common/loader';
 
-export default function ConfigModel({ handleOpen, handleClose, handleParentModalClose, providerConf, customProvider, onSelectedProvider, onLoadProvider, proxyObj }: { handleOpen: boolean, handleClose: () => void, handleParentModalClose: () => void, providerConf?: ProviderI | null, customProvider: ProviderI[], onSelectedProvider: any, onLoadProvider: () => void, proxyObj: restImportConfigI }) {
+export default function ConfigModel({ handleOpen, handleClose, handleParentModalClose, providerConf, proxyObj }: { handleOpen: boolean, handleClose: () => void, handleParentModalClose?: () => void, providerConf?: ProviderI | null, proxyObj: restImportConfigI }) {
+    const dispatch = useDispatch();
     const { t: translate } = useTranslation();
-    const [customProviderList, setCustomProviderList] = useState<ProviderI[]>(customProvider)
+    // const [customProviderList, setCustomProviderList] = useState<ProviderI[]>(customProvider)
```
[the remark this answers is not in the document]
Comment (Ponnarasi-TechConative) [reply src/core/components/ConfigModel.tsx]: commended line removed

File: src/core/components/ConfigModel.tsx (hunk 1 above)
Comment (Ponnarasi-TechConative) [reply src/core/components/ConfigModel.tsx]: Yes Vivek. Changed domain name from configuration

File: src/core/components/appStore/Slice.tsx (hunk 4 above)
Comment (Ponnarasi-TechConative) [reply src/core/components/appStore/Slice.tsx:12]: Slice is used to access and manage data across multiple components.

File: src/core/components/ConfigModel.tsx (hunk 2 above)
Comment (Ponnarasi-TechConative) [reply src/core/components/ConfigModel.tsx]: Changed domain name from configuration

File: src/core/components/ConfigModel.tsx (hunk 3 above)
Comment (Ponnarasi-TechConative) [reply src/core/components/ConfigModel.tsx]: Changed domain name from configuration

**Review by vivek-dhayalan — APPROVED** [review]

**Pull request merged**